tests/testthat/test-helpers_knitr_NextBest.R

```diff
@@ -0,0 +1,503 @@
+        "target DLT rate, either during [0.35] or at the end of the trial ",
+        "[0.3]."
+      )
+    )
+  )
+})
+
+test_that("knit_print.NextBestMaxGainSamples works correctly", {
+  x <- .DefaultNextBestMaxGainSamples()
+  expect_equal(
+    knit_print(x, asis = FALSE),
+    paste0(
+      "The dose recommended for the next cohort will be the one which is ",
+      "closest to Gstar, the dose for which the full Bayes posterior estimate ",
+      "of the probability of toxicity maximises the gain relative to the ",
+      "target toxicity rate, either during [0.35] or at the end of the trial [0.3]."
+    )
+  )
+  expect_equal(
+    knit_print(
+      x,
+      asis = FALSE,
+      tox_label = "DLT"
+    ),
+    paste0(
+      paste0(
+        "The dose recommended for the next cohort will be the one which is ",
+        "closest to Gstar, the dose for which the full Bayes posterior estimate ",
+        "of the probability of DLT maximises the gain relative to the target ",
+        "DLT rate, either during [0.35] or at the end of the trial [0.3]."
+      )
+    )
+  )
+})
+
+test_that("knit_print.NextBestOrdinal works correctly", {
+  x <- .DefaultNextBestOrdinal()
+  expect_equal(
+    knit_print(x, asis = FALSE),
+    paste0(
+      "Based on a toxicity grade of 1: The dose level recommended for the ",
+      "next cohort will be selected as follows:\n\n-  First, the 25th centile ",
+      "of the posterior distribution of toxicity will be calculated for all ",
+      "dose levels that are eligible according to the  Increments rule.\n",
+      "-  Next, the \"target dose\" (which may not be part of the dose grid) ",
+      "for which the 25th centile of the posterior distribution of toxicity is ",
+      "exactly equal to the target rate of 0.25 will be determined.\n",
+      "- Finally, the dose level whose absolute distance from the target dose ",
+      "is smallest will be selected as the recommended dose for the next cohort\n\n"
+    )
+  )
+  x1 <- NextBestOrdinal(grade = 2L, rule = NextBestMTD(target = 0.25, derive = x@rule@derive))
+  expect_equal(
+    knit_print(
+      x1,
+      asis = FALSE,
+      tox_label = "DLT",
+      target_label = "a custom measure"
+    ),
+    paste0(
+      "Based on a toxicity grade of 2: The dose level recommended for the ",
+      "next cohort will be selected as follows:\n\n-  First, a custom ",
+      "measure of the posterior distribution of DLT will be calculated for all ",
+      "dose levels that are eligible according to the  Increments rule.\n",
+      "-  Next, the \"target dose\" (which may not be part of the dose grid) ",
+      "for which a custom measure of the posterior distribution of DLT is ",
+      "exactly equal to the target rate of 0.25 will be determined.\n",
+      "- Finally, the dose level whose absolute distance from the target dose ",
+      "is smallest will be selected as the recommended dose for the next cohort\n\n"
+    )
```

Comment:
these tests look good, thanks, but do we need all the new snapshots? at least in GitHub they are not useful because I just see the raw html code and not the page or contents. and I thought in the last PR we did not add any snapshots

Reply:
Ah, ok, then we can keep this as is.